tests/models/test_hooks.py

```diff
@@ -301,14 +306,14 @@ def _train_batch(trainer, model, batches):
                 dict(name='on_before_zero_grad', args=(ANY, )),
                 dict(name='optimizer_zero_grad', args=(0, i, ANY, 0)),
                 # TODO: `on_before_backward`
-                dict(name='backward', args=(ANY, ANY, 0)),
+                *([dict(name='backward', args=(ANY, ANY, 0))] if kwargs.get('plugins') != 'deepspeed' else []),
```

Comment:
@SeanNaren I'm assuming this is correct as deepspeed handles bwd?

Reply:
I think I understand what's happening here, if the scenario is why the LightningModule hook isn't being called, yes it's handled by Deepspeed

Reply:
>  if the scenario is why the LightningModule hook isn't being called

correct

> yes it's handled by Deepspeed

Do you think we should still call the `backward` hook in this case if the user has overridden it?